sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PointWriter.scala

```diff
@@ -332,6 +336,65 @@ class PointWriter(container: CosmosAsyncContainer,
   }
   // scalastyle:on return
 
+  // scalastyle:on multiple.string.literals
+  private def patchBulkUpdateWithRetry(
+                                      partitionKeyValue: PartitionKey,
+                                      objectNode: ObjectNode): Unit = {
+
+      var exceptionOpt = Option.empty[Exception]
+      val patchBulkUpdateOperation = PatchBulkUpdateOperation(taskDiagnosticsContext,
+          CosmosItemIdentifier(objectNode.get(CosmosConstants.Properties.Id).asText(), partitionKeyValue))
+
+      for (attempt <- 1 to cosmosWriteConfig.maxRetryCount + 1) {
```

Comment:
why not 0 to maxRetryCount and save the whole + 1 magic?

Reply:
But even there without the + 1 magic it would perfectly ok, 

for (attempt <- 1 to cosmosWriteConfig.maxRetryCount + 1) {
log.logWarning( s"patch update item $patchBulkUpdateOperation attempt #$attempt max remaining retries " + s"${cosmosWriteConfig.maxRetryCount + attempt}, encountered ${e.getMessage}")

Reply:
Only diff would be that the pound=prefixed index starts 0 absed?

Also a NIT by the way

Reply:
updated. the same pattern also used for other retry methods, so updated there as well